### ISSUE TITLE
Feature Request: GD TASBot Parser

### ISSUE THREAD

**Issue by 68r** [TASVideos/tasvideos#1505]
can you guys please consider adding a parser for (gd tasbot)[https://www.youtube.com/watch?v=o1SfCT0Vrvg]? its a tas tool for geometry dash thats used by a lot of people.

saved attempts in a level can be used as 'rerecords' and in game time is recorded once you finish a level.

i guess you can see how the tool modifies the game from their 'toastedmarshmallow.dll' file you get upon unzipping, but im really not sure. heres (their discord)[https://discord.com/invite/HMavzxs9je] if you want to ask anything.

**Comment by dwangoac** [comment]
As a quick note, I had a chat with the author of the tool and we came to an agreement that their use of the term "TASBot" on its own was confusing. After discussing at length the author agreed to a change to the name GDTASBot which ended up becoming GD TASBot in print and in their Discord name which wasn't quite what I had in mind given the possible search engine confusion but is at least a distinct enough name. If the parser is added, any TAS's with it should be clearly labeled as belonging to the Geometry Dash project to avoid (further) confusion.

**Comment by YoshiRulz** [comment]
~~Toasted Marshmallow is a much better name~~

As someone unfamiliar with Geometry Dash, is there a set of "campaign" levels, or is it all user-generated? There have been TASes published for user-generated levels, SMW hacks for example, but I believe it would be a first to have movies in a bespoke format AND for a game that's entirely user-generated content.

**Comment by zzthebooplister** [comment]
> ~Toasted Marshmallow is a much better name~
> 
> As someone unfamiliar with Geometry Dash, is there a set of "campaign" levels, or is it all user-generated? There have been TASes published for user-generated levels, SMW hacks for example, but I believe it would be a first to have movies in a bespoke format AND for a game that's entirely user-generated content.

yes, there is a set of campaign levels, those being gd's 22 main levels.

**Comment by EnderNon** [comment]
> > ~Toasted Marshmallow is a much better name~
> > As someone unfamiliar with Geometry Dash, is there a set of "campaign" levels, or is it all user-generated? There have been TASes published for user-generated levels, SMW hacks for example, but I believe it would be a first to have movies in a bespoke format AND for a game that's entirely user-generated content.
> 
> yes, there is a set of campaign levels, those being gd's 22 main levels.

you can only really tas Platformer levels for speedrunning though, since all other levels move at the exact same speed.

**Comment by 1xxxxxxxxx** [comment]
> ~Toasted Marshmallow is a much better name~
> 
> As someone unfamiliar with Geometry Dash, is there a set of "campaign" levels, or is it all user-generated? There have been TASes published for user-generated levels, SMW hacks for example, but I believe it would be a first to have movies in a bespoke format AND for a game that's entirely user-generated content.

There exists The Tower, 4 levels in the first "level", other levels will be added soon
